### PR TITLE
Attempt to fix _some_ issues

### PR DESCRIPTION
**Pull request by joelthe1**
⚠️ Though this PR makes the code 'run', it does not train a model successfully.
## Notes of changes made
- Moved the directory creation outside of an `if` condition that did not predictably trigger.
- Added some `max_f1` logic
- Made the `train()` return more variables
- *Swapped order of `scheduler.step()`  and `optimizer.step()` based on a warning message I saw.* Please verify for yourself if this makes sense.
- Fixed the key name of the `f1` return value.

### CODE REVIEW

**Comment by joelthe1** [comment]
This PR has become stale.
